scripts/storyteller

```diff
@@ -28,31 +30,35 @@ reference_file = '/tmp/storyteller_time_reference'
 
 def exec_cmd(cmd):
     # Use universal_newlines (instead of text) so that this tool can work with any python versions.
-    out = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True, universal_newlines=True)
+    out = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
     stdout, stderr = out.communicate()
     return out.returncode, stdout, stderr
 
 
 def build_options(after=0, before=0, context=0):
     options = []
     if after:
-        options.append('-A {}'.format(after))
+        options += ['-A', str(after)]
     if before:
-        options.append('-B {}'.format(before))
+        options += ['-B', str(before)]
     if context:
-        options.append('-C {}'.format(context))
+        options += ['-C', str(context)]
 
-    return ' '.join(x for x in options)
+    return options
 
 
 def find_log(logpath, log, regex, after=0, before=0, context=0, field=0):
     options = build_options(after, before, context)
     if field <= 0:
-        cmd = 'find -L {}/{}* -newer {} | xargs ls -rt | xargs zgrep -a {} "{}"'.format(logpath, log, reference_file, options, regex)
+        cmd0 = ['find', '-L'] + glob.glob("{}/{}*".format(logpath, log)) + ["-newer", reference_file]
```

Comment:
>glob.glob [](http://example.com/codeflow?start=32&length=9)

No need to glob?

Reply:
need to glob, otherwise the program's failed: `find: ‘/var/log/syslog*’: No such file or directory`

Reply:
I am worried about super long cmd args list. How about `find /var/log/ -name 'syslog*'` which does not require globing?

Reply:
I observed there's not much significant difference b/w the perf of the original code and the new code using glob even super long list. But your approach also works, fixed the code not to use glob.

Reply:
Thanks for the fixing! To clarify, I am worried about super long cmd args, which may above some shell limit. I am not talking about performance.